docs: transform addons (#74379)
metabase/metabase: docs/data-studio/transforms/query-transforms.md

The reason transform variables must have a default value (or be optional) is that transforms run on a schedule, so there's no way to pass a value to the variable when the job runs the transform.

> The incremental `{%raw%}[[WHERE id > {{checkpoint}}]]{% endraw %}` pattern shown in [Incremental query transforms](#incremental-query-transforms) is an example of this an optional variable in practice. See also [optional variables](../../questions/native-editor/optional-variables.md).

Author: No longer applicable

1. Go to the transform's page in **Data studio > Transforms**.
2. Switch to **Settings** tab.
3. In **Column to check for new values**, select the column in one of the source tables that Metabase should check to determine which values are new. Only some columns are eligible. See [prerequisites for incremental transforms](./transforms-overview.md#prerequisites-for-incremental-transforms).

> ## Convert models to transforms

Author: Moved from transforms overvie

> > {% include plans-blockquote.html feature="Converting models to transforms"%}

Reviewer: ```suggestion
{% include plans-blockquote.html feature="Converting models to transforms"%}
```

3. Replaces every question, dashboard, and other item that used the model with the transform's output table.
4. Converts the original model to a question.

You must be an admin to convert models, and the model's database must [support transforms](transforms-overview.md#databases-that-support-transforms).

To convert a model into a transform:

1. Review [Replacing data sources](../dependencies/replace-data-sources.md) docs for overview and limitations of the process.
2. Open **Data Studio** and select **Transforms** in the sidebar.
3. Click **Tools > Migrate models**.
4. Find the model you want to convert and click it to open its details panel. The panel shows the model's name, database, and collection, and a list of items that depend on it.
5. Click **Convert to a transform**.
6. Fill out the transform settings. See [Create a transform](#create-a-query-based-transform) for the overview of settings.
7. Click **Convert to a transform**.

Metabase runs the conversion in the background. A status indicator at the bottom of the screen shows progress.

If the transform run fails, Metabase stops and leaves the model unchanged—nothing is replaced.

> If the transform runs successfully but the source swap fails afterward, the transform and its output table are kept. The model is left unchanged. You can complete the migration manually using [Replace data sources](../dependencies/replace-data-sources.md) to point remaining content from the model to the transform's output table.

Reviewer: ```suggestion
If the transform runs successfully, but the source swap fails afterward, the transform and its output table are kept. The model is left unchanged. You can complete the migration manually using [Replace data sources](../dependencies/replace-data-sources.md) to point the remaining content from the model to the transform's output table.
```

> Once conversion completes, all content that previously queried the model now queries the transform's output table, and the model becomes a saved question.

Reviewer: ```suggestion
Once conversion completes, all content that previously queried the model now queries the transform's output table. The original model just becomes a question.
```
